feature_engine/model/model_builder.py

```diff
@@ -366,18 +366,31 @@ def create_holdout_sets(self, val_size: float=0.1, test_size: float=None) -> Non
         """
         # Calls the `define_dataset_for_model()` function to get the `X`s and the `y`s
         self.X, self.y = self.define_dataset_for_model()
+        
         # If we need to split out a test set
-        if test_size:
+        if self.test_dataset_names == None:
```

Comment:
changing this if condition to mean, "if we need to split out a test set, because there is no external test dataset"